### PR TITLE
Feat (tooltip) refactor (popover) to base it off tooltip

### PR DESCRIPTION
**Pull request by mathsusan**
Adding tooltip to angular library

## Description
refactored popover to base it off tooltip 

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [ x] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/tooltip/tooltip.directive.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+  /** @prop Sets direction of the tooltip right,left,top, or bottom */
+  @Input() direction: tooltipDirection = 'top';
+
+  /** @prop Sets delay for before closing the tooltip */
+  @Input() delay: number = 500;
+
+  /** @prop Sets the tooltip trigger MouseEnter, Click or Focus - MouseEnter is default*/
+  @Input() tooltipTrigger: string = 'MouseEnter';
+
```
Comment (daanishnasir) [inline angular/src/lib/tooltip/tooltip.directive.ts]: offset need comment description?  /**@prop

Comment (mathsusan) [reply angular/src/lib/tooltip/tooltip.directive.ts]: added :-)

File: angular/src/lib/popover/popover.directive.ts [hunk 2]
```diff
@@ -1,153 +1,35 @@
...
-    }
+  constructor( overlay: Overlay,
+               overlayPositionBuilder: OverlayPositionBuilder,
+               _sso: ScrollStrategyOptions,
+               elementRef: ElementRef) {
+                super(
+                  overlay,
+                  overlayPositionBuilder,
+                  _sso,
+                  elementRef);
```
Comment (bfbiggs) [inline angular/src/lib/popover/popover.directive.ts]: Can we do some formatting here for the closing brackets/parens?

File: angular/src/lib/tooltip/tooltip-container.component.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      );
+    }
+  }
+
+   /** @prop Sets the id for screen reader */
+  @Input() id: string;
+
+    constructor(private elementRef: ElementRef) {
+    }
+
```
Comment (daanishnasir) [inline angular/src/lib/tooltip/tooltip-container.component.ts]: we can remove these empty lines here and close out the constructor {} on 1 line.

File: angular/src/lib/tooltip/tooltip-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  @Input()
+  set lightBackground(lightBackground: boolean) {
+    if ( lightBackground ) {
+      // md-popover has the light background
+      this.elementRef.nativeElement.children[0].classList.remove(`md-tooltip`);
+      this.elementRef.nativeElement.children[0].classList.add(`md-popover`);
+    }
+  }
+
+  /** @prop Sets whether or not to show the arrow on the tooltip  */
```
Comment (daanishnasir) [inline angular/src/lib/tooltip/tooltip-container.component.ts]: comment looks indented

File: angular/src/lib/tooltip/tooltip-container.scss [hunk 5]
```diff
@@ -0,0 +1,61 @@
+.md-tooltip {
+    &.md-event-overlay {
+      position: static;
+      > .md-event-overlay__children {
+        position: static;
+        padding: 12px;
+      }
+
+       > .md-event-overlay__arrow {
+    //     z-index: $zindex-top-bar-fixed - 3;
```
Comment (daanishnasir) [inline angular/src/lib/tooltip/tooltip-container.scss]: safe to remove the comment?

**Comment by daanishnasir** [comment]
lets squash and rebase @mathsusan

File: angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap [hunk 6]
```diff
@@ -7,33 +7,41 @@ exports[`Popover Test should match snapshot 1`] = `
   <div
     class="cdk-overlay-connected-position-bounding-box"
     dir="ltr"
-    style="top: 0px; left: 0px; height: 0px; width: 0px; align-items: flex-start; justify-content: center;"
+    style="left: 0px; bottom: 0px; height: 0px; width: 0px; align-items: center; justify-content: flex-end;"
```
Comment (ArthusLiang) [inline angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap]: I suggest to reduce the inline style in HTML template as many as possible. Some scrum team will do some security check to remove all the inline style. Using css class will be a better choice. And you can still use Javascript to change the dom style dynamically.

File: angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap [hunk 7]
```diff
@@ -7,33 +7,41 @@ exports[`Popover Test should match snapshot 1`] = `
...
   >
     <div
       class="cdk-overlay-pane"
       id="cdk-overlay-0"
-      style="pointer-events: auto; position: static; transform: translateX(5px);"
```
Comment (ArthusLiang) [inline angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap]: Translate is a good choice!!! Like it! Position (left/top/right/bottom) is not convenient to use in CDK overlay. The overlay bounding makes position complex.

File: angular/src/lib/tooltip/tooltip-container.component.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
+import { Component, Input,  TemplateRef, ElementRef} from '@angular/core';
+
+
+
+@Component({
+  selector: 'md-tooltip-container',
+  styleUrls: ['tooltip-container.scss'],
+  template: `
+  <div class="md-tooltip md-event-overlay md-event-overlay--arrow">
```
Comment (ArthusLiang) [inline angular/src/lib/tooltip/tooltip-container.component.ts]: I am not sure if you can build the class="md-tooltip md-event-overlay md-event-overlay--arrow" to the host.  The host here seems to be a tag without any attribute.

File: angular/src/lib/tooltip/tooltip-container.component.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    this.elementRef.nativeElement.children[0].classList.add(`md-event-overlay--${direction}`);
+  }
+
+  /** @prop Sets a background color for the tooltip | null */
+  @Input()
+  set lightBackground(lightBackground: boolean) {
+    if ( lightBackground ) {
+      // md-popover has the light background
+      this.elementRef.nativeElement.children[0].classList.remove(`md-tooltip`);
+      this.elementRef.nativeElement.children[0].classList.add(`md-popover`);
```
Comment (ArthusLiang) [inline angular/src/lib/tooltip/tooltip-container.component.ts]: I'd prefer to use a variable store this.elementRef.nativeElement.children[0] the first time I use it. Dom operation uses a lot of cpu resource.

File: angular/src/lib/tooltip/tooltip-container.component.ts (hunk 8 above)
Comment (daanishnasir) [reply angular/src/lib/tooltip/tooltip-container.component.ts]: Yeah, add these classes to the host so it we can take out the div wrapper.

File: angular/src/lib/tooltip/tooltip-container.component.ts (hunk 9 above)
Comment (mathsusan) [reply angular/src/lib/tooltip/tooltip-container.component.ts]: took this out and refactored

File: angular/src/lib/tooltip/tooltip-container.component.ts (hunk 8 above)
Comment (mathsusan) [reply angular/src/lib/tooltip/tooltip-container.component.ts]: refactored how I was adding these classes .. host didn't worked but found a better way

File: angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap (hunk 7 above)
Comment (mathsusan) [reply angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap]: lol .. I didn't do this the cdk did

File: angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap (hunk 6 above)
Comment (mathsusan) [reply angular/src/lib/popover/tests/__snapshots__/popover.component.spec.ts.snap]: this again is the cdk ..

File: angular/src/lib/popover/popover.directive.ts [hunk 10]
```diff
@@ -1,153 +1,35 @@
...
+export class PopoverDirective extends TooltipDirective {
 
   /** @prop Sets content in the popover can be a string or a template */
   @Input('mdPopover') content: string | TemplateRef<any>;
 
-  /** @prop Sets direction of the popover right,left,top, or bottom */
-  @Input() direction: popoverDirection = 'right';
-
   /** @prop shows the arrow or not */
   @Input() showArrow: string = 'true';
```
[the remark this answers is not in the document]
Comment (mathsusan) [reply angular/src/lib/popover/popover.directive.ts]: yes that would be much better .. corrected that

**Pull request merged**